polygeist/tools/cgeist/Lib/ValueCategory.cc

```diff
@@ -758,3 +758,53 @@ ValueCategory ValueCategory::Xor(mlir::OpBuilder &Builder, mlir::Location Loc,
                                  mlir::Value RHS) const {
   return IntBinOp<arith::XOrIOp>(Builder, Loc, val, RHS);
 }
+
+ValueCategory ValueCategory::Insert(OpBuilder &Builder, Location Loc, Value V,
```

Comment:
Would be good to have comments for all these functions (in the header file).

Reply:
You're right. As this involves all of the functions in `ValueCategory.h`, I'd rather add comments for every function in a different PR.